src/cr/cube/cubepart.py

```diff
@@ -756,6 +752,11 @@ def table_std_dev(self):
     def table_std_err(self):
         return np.array([row.table_std_err for row in self._matrix.rows])
 
+    @lazyproperty
+    def unweighted_counts(self):
+        """2D tuple of unweighted counts."""
```

Comment:
I guess the subsequent commits will introduce similar change for other properties. However, I'm not sure that "2D tuple" is clear enough in terms of what we actually put in. If it's a 2D `ndarray`, it's clear by definition, because the concept of a dimension is very well defined there, and very common. Not as much in the tuple world, afaict. I'll get back to this after I review the whole thing. I guess the main motivation behind this change is to have an unmutable type returned, it that right?

Reply:
I hadn’t realized this. It makes some sense, but a little surprised that the requester of this property doesn’t expect an ndarray?

Reply:
I guess I figure one of the design principles for the `cr.cube` public interface is that users of this library should have no direct need for array operations since they are using the results for rendering or reporting types of purposes. An example would be whaam, who uses it via the SOA interface. Everything the library can express can be expressed as primitives that could, for example, be directly serialized to JSON. Whatever whaam gets from SOA, it's not ndarrays once it gets over there.

I haven't thought that through 100%, but I have noticed that array types and operations have worked their way into `exporter` code where they don't "natively" belong. We shouldn't have to do any additional analysis operations like dividing one array by another and if the ndarray-ness of a returned value needs to be taken advantage of, I'm thinking that represents work that probably belongs in `cr.cube` (and there is at least a couple of examples of that I can think of).

Now I don't feel strongly enough about this to propose we convert all the external interface to return Python primitives right now, but I think the principle is probably pretty correct. Anyway, that's what I had in mind with this.

Aside from its mutability, an ndarray has some surprising behaviors to someone who's not familiar and I can't think of a good reason that clients of `cr.cube` should have to be familiar with `numpy` (or anything outside native Python) to be able to use it effectively.